src/ucp/tag/tag_send.c

```diff
@@ -152,13 +162,13 @@ ucp_tag_send_inline(ucp_ep_h ep, const void *buffer, size_t count,
 
     length = ucp_contig_dt_length(datatype, count);
 
-    if ((ssize_t)length <= ucp_ep_config(ep)->tag.max_eager_short &&
-        ucp_memory_type_cache_is_empty(ep->worker->context)) {
+    if (ucp_tag_eager_is_inline(ep, &config->tag.max_eager_short, length)) {
```

Comment:
can you pls pass &ucp_ep_config(ep)->tag.max_eager_short to the function and not cache config in local var?
i tried to experiment with it in the past, and this way perf was slightly better (probably could spill less registers)
same for line 170